gr-digital/lib/constellation.cc

```diff
@@ -269,10 +319,10 @@ std::vector<float> constellation::calc_soft_dec(gr_complex sample, float npwr)
     for (int i = 0; i < M; i++) {
         // Calculate the distance between the sample and the current
         // constellation point.
-        float dist = std::abs(sample - d_constellation[i]);
+        float dist = powf(std::abs(sample - d_constellation[i]), 2.0f);
```

Comment:
Probably this generates better asm code, avoiding to compute a square root and then `powf`.
```suggestion
        const gr_complex z = sample - d_constellation[i];
        float dist = z.real() * z.real() + z.imag() * z.imag();
```